### ISSUE TITLE
Markdown Preview not working with 0.14

### ISSUE THREAD

**Issue by Gander7** [foambubble/foam#717]
Updated (and happy) with 0.14 except that using "Markdown: Open Preview" and "Markdown: Open Preview to side" opens a blank tab. I'm expecting it to render the markdown file that I had open when I ran the command.

I'm relatively new to debugging extensions, are the foam logs I can use to debug the issue for either foam or for markdown?

**Comment by MaxBalashov** [comment]
Same problem!

**Comment by pderaaij** [comment]
To look into this I need a bit more info. Does it happen with all files? Is it possible to share a markdown file? There could be a clue in the Foam log that would be helpful to have.

**Comment by RotationMatrix** [comment]
Here's the minimal steps to reproduce from my experience.

1. Create a new file and set editor language to Markdown.
2. Paste `[link](a)`.
3. Open Markdown Preview.

And the error from "Log (Extension Host)" Output (traceback truncated). This looks like the only related error in Output logs.

```
[2021-07-14 07:07:46.930] [exthost] [error] [vscode.markdown-language-features] provider FAILED
[2021-07-14 07:07:46.930] [exthost] [error] TypeError: Cannot convert undefined or null to object
	at Function.keys (<anonymous>)
	at Array.<anonymous> (.vscode-oss/extensions/foam.foam-vscode-0.14.0/out/features/preview-navigation.js:74:16)
	at o.tokenize (/Applications/VSCodium.app/Contents/Resources/app/extensions/markdown-language-features/dist/extension.js:1:198273)
```

I'm still digging for the exact source of the error but it appears this could be a side effect of [4db8c559](https://github.com/foambubble/foam/commit/4db8c5596919ed580395b40cdc95312ed555c9d5).

**Comment by pderaaij** [comment]
I suspect a null check is missing which is causing this. Will try to verify and fix it tonight.

**Comment by break7533** [comment]
The following example should be useful checking some examples from the feedback in the discord.


```md

New File example: This is an example! 

This file & this file crash.

Split - case = 100% error rate
```

**Comment by Saskia-R** [comment]
Attached is a very minimal file that contains all characters or elements identified so far in the discord server, that break the preview. I included some elements using the recommended extension Markdown All in one. 

[exampleFile.md](https://github.com/foambubble/foam/files/6816658/exampleFile.md)

**Comment by pderaaij** [comment]
Sorry about this, unexpected situation which I could have handled a bit more defensively in the code. Anyway, PR has been created and hopefully, we can release a fix soon.

**Comment by Gander7** [comment]
I would like to add the character @ to the list :)
Thanks @Saskia-R , did some of the work to see you had as well :)
Thanks @pderaaij for the quick turnaround :)

**Comment by pderaaij** [comment]
The problem is that the null pointer issues causes the tokenisation to stop. As a result, it looks like those characters are the issue, but it is actually triggered by the issue before. Thanks for all help and reference files. I'll use them to create some additional tests.

**Comment by riccardoferretti** [comment]
fixed by #718
